src/Symfony/Component/Security/Guard/Authenticator/AbstractFormLoginAuthenticator.php

```diff
@@ -41,9 +41,14 @@
      * login page directly), this returns the URL the user should be redirected
      * to after logging in successfully (e.g. your homepage).
      *
+     * @deprecated Implement onAuthenticationSuccess() instead of needing this function.
+     *
      * @return string
      */
-    abstract protected function getDefaultSuccessRedirectUrl();
+    protected function getDefaultSuccessRedirectUrl()
```

Comment:
As this one was abstract, it was always implemented, so this code is never going to be executed, right?